src/marqo/vespa/vespa_client.py

```diff
@@ -563,6 +577,54 @@ async def _feed_batch_async(self, batch: List[VespaDocument],
 
         return FeedBatchResponse(responses=responses, errors=errors)
 
+    async def _update_batch_async(self, batch: List[VespaDocument],
+                                  schema: str,
+                                  connections: int, timeout: int) -> UpdateBatchResponse:
+        async with httpx.AsyncClient(limits=httpx.Limits(max_keepalive_connections=connections,
+                                                         max_connections=connections)) as async_client:
+            semaphore = asyncio.Semaphore(connections)
+            tasks = [
+                asyncio.create_task(
+                    self._update_document_async(semaphore, async_client, document, schema, timeout)
+                )
+                for document in batch
+            ]
+            await asyncio.wait(tasks, return_when=asyncio.ALL_COMPLETED)
+
+        responses = []
+        errors = False
+        for task in tasks:
+            result = task.result()
+            responses.append(result)
+            if result.status != 200:
+                errors = True
+
+        return UpdateBatchResponse(responses=responses, errors=errors)
+
+    async def _update_document_async(self, semaphore: asyncio.Semaphore, async_client: httpx.AsyncClient,
+                                     document: VespaDocument, schema: str,
+                                     timeout: int) -> UpdateBatchDocumentResponse:
+        doc_id = document.id
+        data = {'fields': document.fields}
+
+        async with semaphore:
+            end_point = f'{self.document_url}/document/v1/{schema}/{schema}/docid/{doc_id}?create=false'
+            data["condition"] = f'{schema}.marqo__id==\"{doc_id}\"'
```

Comment:
pydocs of the public method should indicate that will fail if doc ID doesn't exist (and error code)